File: app/src/main/java/com/firemaples/getuitest/CoreApplication.java
package com.firemaples.getuitest;

import android.app.Application;
import com.igexin.sdk.PushManager;

public class CoreApplication extends Application {
    private Logger logger = LoggerFactory.getLogger(CoreApplication.class);

    @Override
    public void onCreate() {
        super.onCreate();

        logger.debug("onCreate()");

        PushManager.getInstance().initialize(this, GetuiPushService.class);
        PushManager.getInstance().registerPushIntentService(this, GetuiMessageService.class);

        PushManager.getInstance().setPrivacyPolicyStrategy(this, true);
//        PushManager.getInstance().turnOnPush(this);
//        PushManager.getInstance().bindAlias(this, "abcd");
//        PushManager.getInstance().setSilentTime(this, 0, 23);
    }
}

File: app/src/main/java/com/firemaples/pushtest/CoreApplication.kt
package com.firemaples.pushtest

import android.app.Application
import com.firemaples.pushtest.LoggerFactory.Companion.getLogger

class CoreApplication : Application() {
    private val logger: Logger = getLogger(CoreApplication::class.java)

    override fun onCreate() {
        super.onCreate()
        logger.debug("onCreate()")

        PushManager.init(this)
    }
}